### ISSUE TITLE
0.59.x Discussion

### ISSUE THREAD

**Issue by grabbou** [react-native-community/react-native-releases#79]
Conversation on this thread are limited to 0.59 releases, major issues and backport (cherry-pick) requests from commits that are already on master.

An example of a good such request is a bug fix for a serious issue that has been merged into master but did not make the 0.59 RC cut.

In other words, if you cannot point to a particular commit on master, then your request likely belongs as a new issue in http://github.com/facebook/react-native/issues.

**Comment by grabbou** [comment]
We haven't cut 0.59-stable branch yet. The latest release candidate is 0.58.0-rc.3. You can track it's progress here. This thread is to inform you when next release is going to happen.

**Comment by grabbou** [comment]
This is the first release to ship with the extracted CLI. The release is generally ready to be cut and released, but we are waiting for the last few commits that are missing to make the new CLI work with React Native.

I am going to give you an update in the next few days.

**Comment by kelset** [comment]
This version will also *likely*:
- have the new JSC
- not have WebView

(TBC)

**Comment by grabbou** [comment]
CLI commit is shipped: https://github.com/facebook/react-native/commit/1af390be19000b2756fc646e72f152dd383a0bb1. Waiting for tests to be fixed and we are cutting it.

**Comment by yurykorzun** [comment]
Will it include the latest Yoga update? https://github.com/facebook/react-native/tree/master/ReactCommon/yoga/yoga

I have a reason to believe the old one causes some issues in my hybrid app, would be nice get an update.

**Comment by kelset** [comment]
@yurykorzun when the branch will be cut it will be a 'replica' of everything on master up to that point, so my guess is yes ;)

**Comment by eviltandem** [comment]
we're close to cutting the release on the apis

**Comment by grabbou** [comment]
Yes, @kelset is right. I am finalising 0.58 right now and will work on making 0.59.0-rc.0 tomorrow. I will keep you updated.

**Comment by Eyesonly88** [comment]
Any chances of landing Hooks in this release?

**Comment by kelset** [comment]
@Eyesonly88 the basic answer is that *it depends*, we need to wait on a commit that syncs React in the codebase. If you look at the commit history on master you'll see some commits from the FB team called "React sync", so keep an eye out for those.

We don't have an estimate yet, so don't hold your breath - we'll do what we can 💪

**Comment by yurykorzun** [comment]
I don't mind hooks to be added later and 0.59 RC being released as soon as possible

**Comment by hindsc52** [comment]
> the possibility of cutting a parallel 0.60 RC as soon as React w/ Hooks lands in master

This would be fantastic. Would help with development of many projects that want to start using hooks too.

**Comment by ithustle** [comment]
Any chances to add a native audio api?

**Comment by grabbou** [comment]
Waiting for the few last tests to be fixed and we are making this release. Please check https://github.com/facebook/react-native/issues/23108 if you want to track progress when it happens or help us getting it fixed!

**Comment by matthargett** [comment]
Even if 0.59 doesn't get Hooks upon initial release, it won't be "small" in terms of value -- the JSC upgrade, resulting cut polyfills, CLI, and other cleanup work is pretty huge.

Just to state my preference here, though: my preference is to get master CI stable before cutting 0.59-rc, and for 0.59 to include Hooks. I'd rather have a longer-supported 0.59 than an immediate 0.60 follow-up. I agree that it depends on what the React sync looks like in terms of complexity. If it would be possible for @hramos @yungters to do it against the React master githash in a PR (rather than direct commit from phabricator), that might help get an idea earlier.

**Comment by kanalasumant** [comment]
The later the better, worth waiting for

**Comment by gtebbutt** [comment]
@kanalasumant Not for those of us waiting for other fixes that didn't make it into 0.58!

**Comment by matthargett** [comment]
@gtebbut you can request cherry picks of fixes on the 0.58 issue in this repo. 0.57 had 8 patch releases, which allowed users with production apps to keep on a stable base while getting critical fixes (and even some quality of life improvements like a React upgrade). If there’s specific commits you’re interested in getting merged to 0.58, comment there.

**Comment by hramos** [comment]
@matthargett how would the PR help? Is this to gauge how complex the sync commit would be, in order to make the decision whether it's preferred to make the cut before the commit actually lands on master?

Happy to adjust as needed, we do have tooling to seamlessly export our internal diffs to a GitHub PR so this is doable, I can coordinate with Tim.

**Comment by matthargett** [comment]
Yes, to gauge how complex the sync commit would be. If it has a bunch of ripple effects, then we know for sure not to hold up 0.59. I’d only do this if the folks who really want to get 0.59 cut immediately without a passing CI can’t wait.

**Comment by objectiveSee** [comment]
We are waiting for a fix in `master` regarding a crash in `RCTSRWebSocket`, so our vote is to get 0.59 and use 0.60 for Hooks like @grabbou suggested. 

https://github.com/facebook/react-native/pull/22439#issuecomment-450773760

**Comment by sijad** [comment]
I also vote for a longer-supported 0.59 with hooks, waiting for more two weeks worth it.

**Comment by grabbou** [comment]
Looks like React sync is around week away, so the general attitude towards the core team was to wait for the sync before releasing.

@objectiveSee, would cherry-picking it into 0.58 do it for you? Looks like the commit itself should be easy to apply to that branch.

**Comment by akshetpandey** [comment]
The linked issue seems like a blocker for 0.59 release with the new JSC.

https://github.com/facebook/react-native/issues/23259

**Comment by eladcandroid** [comment]
React 16.8 is out!
https://reactjs.org/blog/2019/02/06/react-v16.8.0.html

**Comment by kelset** [comment]
👋 everyone.
Yes, the new version of React is out now - the FB team is working on doing the React sync commit so that we can then finally cut 0.59 rc once said commits land (we reached a decision of having Hooks in 59).

As we mentioned times and times again, it will take some time.

**Comment by paulreimer** [comment]
Is this still the best approach for us to try RN+Hooks on a custom RN build for now?
https://github.com/facebook/react-native/issues/21967#issuecomment-434113687

**Comment by grabbou** [comment]
Nothing changed here. This issue is the best place to track using hooks
with React Native right now. The React sync is already in progress - the
0.59 is going to be out really soon.

**Comment by Momazo7u7** [comment]
Finally , been waiting so long , with hooks and reactn we can manage global state finally wihtout using redux :"U

**Comment by kolohelios** [comment]
@Momazo7u7 I can't say I've felt there has been any sort of Redux monopoly for managing state; I've used both Unstated, and more recently, XState with custom context providers to manage application state. And there are many others, like Mobx and Mobx-Statetree, to name a couple more. To your point, however, it will make it easier and less verbose to accomplish the same using hooks. Reducing the need for HoCs is definitely worthy of celebration!

**Comment by salah-ghanim** [comment]
will [facebook/react-native#23278](https://github.com/facebook/react-native/pull/23278) be part of the next release? it's can reduce a lot of unnecessary refactors we need to port our native functionality to react.

**Comment by grabbou** [comment]
@salah-ghanim - everything that is merged to master at the time of the branch cut is going to be part of next release. We haven't cut the 0.59-stable branch yet, so the answer is - yes.

**Comment by grabbou** [comment]
As you know, the [React sync commit](https://github.com/facebook/react-native/commit/ccefc700d0120539eba73747d1d6b65effb0645d) has landed last week on Friday. We have decided to wait for a couple of days to let it settle and detect major regressions before it goes to the majority of developers. The tests are done internally at Facebook (that runs React Native master and constantly deploys to production).

I'll let you know when the 0.59.0-rc.0 is about to be released. Current estimate is around middle of the week when all fixes are merged.

**Comment by dulmandakh** [comment]
@salah-ghanim AppCompatActivity will be a part of 0.59 release. See https://github.com/facebook/react-native/commit/3b9604feda8f9e8fe3dd884912ec7d9be67d7f1d.

- And these might be a breaking change for Kotlin projects, because it add nullability annotations https://github.com/facebook/react-native/commit/c93cbdf1b272cfd60124d9ddb4c52b58ca59d319 and https://github.com/facebook/react-native/commit/b640b6faf77f7af955e64bd03ae630ce2fb09627.
- ReactActivity now extends AppCompatActivity (https://github.com/facebook/react-native/commit/3b9604feda8f9e8fe3dd884912ec7d9be67d7f1d), thus ReactFragmentActivity is deprecated (https://github.com/facebook/react-native/commit/dda2b82a0a49da52b43b50db5a2bda50a216c09b).

**Comment by dulmandakh** [comment]
For brown-field apps these might be a breaking change if they don't extend FragmentActivity, because RN dropped support for plain Activity.

- DatePickerDialogModule supports only FragmentActivity - https://github.com/facebook/react-native/commit/c6c5a173bce3d8c847931d26eddb295956285438
- TimePickerDialogModule supports only FragmentActivity  https://github.com/facebook/react-native/commit/be361d0fc1930b1679c4226e15c1a5b416b94105
- DialogModule supports only FragmentActivity - https://github.com/facebook/react-native/pull/23365. Waiting to be merged.

These changes won't affect green-field apps, because ReactActivity already extends AppCompatActivity, subclass of FragmentActivity.

**Comment by dulmandakh** [comment]
Cherry picks:
- bump Fresco to 1.11, with many security fixes - https://github.com/facebook/react-native/commit/0a071610843475710d54cb2acac4bdb2a190dd59

**Comment by dulmandakh** [comment]
0.59 bumped minimum required version of Java to 8 - https://github.com/facebook/react-native/commit/38eb2a70afa87c49c1e62754f5ae3cd26e7f59c3

**Comment by dulmandakh** [comment]
cherry picks:
- improved Android Network Security config - 
https://github.com/facebook/react-native/commit/5747094532bace3fe6b1ebdf55235e53189baa71
- null annotations for ReactPackage (Kotlin developer experience) - 
https://github.com/facebook/react-native/commit/84f40da990dfd21eb1c21e20f2be0f8b2c5a78e4
- bump targetSdk to 28 - 
https://github.com/facebook/react-native/commit/57f444bd8a175038c367fa1b7d93e2e8ba9de7ed

**Comment by kelset** [comment]
For cherry pick: https://github.com/facebook/react-native/commit/00392ac46b6319dcff2b6df2e5f7bb4ee094612f (just landed)

**Comment by hramos** [comment]
👀 v0.59.0-rc.0 is now on npm

**Comment by grabbou** [comment]
@kelset the requested commit haven't made it to 0.59.0-rc.0. Something to consider for 0.59.0-rc.1

**Comment by Salakar** [comment]
@grabbou I think the default RN config (formerly `require('react-native/local-cli/util/Config');`) might need exposing - it's new home at https://github.com/react-native-community/react-native-cli/blob/master/packages/cli/src/util/loadMetroConfig.js doesn't allow you to access it anymore, some people use this alongside metro's `mergeConfig` functionality, e.g:
- https://github.com/ueno-llc/react-native-starter/blob/master/metro.config.js
- https://github.com/invertase/react-native-firebase/blob/master/tests/metro.config.js

**Comment by Salakar** [comment]
0.59 has a require cycle when debugging:

![image](https://user-images.githubusercontent.com/5347038/52748511-945ed980-2fde-11e9-9e9c-178f0d06351c.png)

`YellowBox.js:67 Require cycle: node_modules/react-native/Libraries/Network/fetch.js -> node_modules/react-native/Libraries/vendor/core/whatwg-fetch.js -> node_modules/react-native/Libraries/Network/fetch.js`

**Comment by Federkun** [comment]
Got an error on android after upgrading: `ERROR: Could not get unknown property 'mergeResourcesProvider' for object of type com.android.build.gradle.internal.api.ApplicationVariantImpl.
`

Any suggestions?

**Comment by mikehardy** [comment]
> Got an error on android after upgrading: `ERROR: Could not get unknown property 'mergeResourcesProvider' for object of type com.android.build.gradle.internal.api.ApplicationVariantImpl. `

> **EDIT**: Solved by following [facebook/react-native#23103 (comment)](https://github.com/facebook/react-native/pull/23103#issue-246733045)

Succinctly for those following: update gradle to 4.10.1 or higher (4.10.3 is current)

For Android devs with experience Gradle 5 used to be incompatible with Gradle 3.3 (the docs said wait till Gradle 3.4) but now the docs say you can go all the way to Gradle 5 if you want though I haven't tested that with react-native - only 4.10.3.

**Comment by Federkun** [comment]
In my own case I had to do the following

```diff
android {
-    compileSdkVersion 26
-    buildToolsVersion "26.0.1"
+   compileSdkVersion 28
```

```diff
- distributionUrl=https\://services.gradle.org/distributions/gradle-4.4-all.zip
+ distributionUrl=https\://services.gradle.org/distributions/gradle-4.10.3-all.zip
```

```diff
    dependencies {
-        classpath 'com.android.tools.build:gradle:3.1.0'
+       classpath 'com.android.tools.build:gradle:3.3.0'
```

**Comment by Kennytian** [comment]
react native 0.59.0-rc.0 can not run 'yarn start'

```
yarn start
yarn run v1.13.0
$ node node_modules/react-native/local-cli/cli.js start
internal/modules/cjs/loader.js:583
    throw err;
    ^
```

**Comment by blainekasten** [comment]
@Kennytian they moved the cli to it's own package: https://github.com/react-native-community/react-native-cli

install that package, and change your start script to look at `node_modules/react-native-cli/index.js start`

**Comment by blainekasten** [comment]
Getting a build error when building in xcode

```
Undefined symbols for architecture x86_64:
  "_JSClassCreate", referenced from:
      facebook::jsc::JSCRuntime::createObject(std::__1::shared_ptr<facebook::jsi::HostObject>)::$_0::operator()() const in libReact.a(JSCRuntime.o)
      facebook::jsc::JSCRuntime::createFunctionFromHostFunction(facebook::jsi::PropNameID const&, unsigned int, std::__1::function<facebook::jsi::Value (facebook::jsi::Runtime&, facebook::jsi::Value const&, facebook::jsi::Value const*, unsigned long)>)::$_1::operator()() const in libReact.a(JSCRuntime.o)
  "_JSContextGetGlobalObject", referenced from:
      facebook::jsc::JSCRuntime::global() in libReact.a(JSCRuntime.o)
      facebook::jsc::JSCRuntime::createFunctionFromHostFunction(facebook::jsi::PropNameID const&, unsigned int, std::__1::function<facebook::jsi::Value (facebook::jsi::Runtime&, facebook::jsi::Value const&, facebook::jsi::Value const*, unsigned long)>)::HostFunctionMetadata::initialize(OpaqueJSContext const*, OpaqueJSValue*) in libReact.a(JSCRuntime.o)
  "_JSEvaluateScript", referenced from:
      facebook::jsc::JSCRuntime::evaluateJavaScript(std::__1::unique_ptr<facebook::jsi::Buffer const, std::__1::default_delete<facebook::jsi::Buffer const> >, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) in libReact.a(JSCRuntime.o)
  "_JSGlobalContextCreateInGroup", referenced from:
      facebook::jsc::JSCRuntime::JSCRuntime() in libReact.a(JSCRuntime.o)
      facebook::jsc::JSCRuntime::JSCRuntime() in libReact.a(JSCRuntime.o)
  "_JSGlobalContextRelease", referenced from:
      facebook::jsc::JSCRuntime::JSCRuntime() in libReact.a(JSCRuntime.o)
      facebook::jsc::JSCRuntime::~JSCRuntime() in libReact.a(JSCRuntime.o)
      facebook::jsc::JSCRuntime::JSCRuntime() in libReact.a(JSCRuntime.o)
  "_JSGlobalContextRetain", referenced from:
      facebook::jsc::JSCRuntime::JSCRuntime(OpaqueJSContext*) in libReact.a(JSCRuntime.o)
  "_JSObjectCallAsConstructor", referenced from:
      facebook::jsc::JSCRuntime::callAsConstructor(facebook::jsi::Function const&, facebook::jsi::Value const*, unsigned long) in libReact.a(JSCRuntime.o)
  "_JSObjectCallAsFunction", referenced from:
      facebook::jsc::JSCRuntime::call(facebook::jsi::Function const&, facebook::jsi::Value const&, facebook::jsi::Value const*, unsigned long) in libReact.a(JSCRuntime.o)
  "_JSObjectCopyPropertyNames", referenced from:
      facebook::jsc::JSCRuntime::getPropertyNames(facebook::jsi::Object const&) in libReact.a(JSCRuntime.o)
  "_JSObjectGetPrivate", referenced from:
      facebook::jsc::JSCRuntime::getHostObject(facebook::jsi::Object const&) in libReact.a(JSCRuntime.o)
      facebook::jsc::JSCRuntime::getHostFunction(facebook::jsi::Function const&) in libReact.a(JSCRuntime.o)
      facebook::jsc::JSCRuntime::createObject(std::__1::shared_ptr<facebook::jsi::HostObject>)::HostObjectProxy::finalize(OpaqueJSValue*) in libReact.a(JSCRuntime.o)
      facebook::jsc::JSCRuntime::createObject(std::__1::shared_ptr<facebook::jsi::HostObject>)::HostObjectProxy::getProperty(OpaqueJSContext const*, OpaqueJSValue*, OpaqueJSString*, OpaqueJSValue const**) in libReact.a(JSCRuntime.o)
      facebook::jsc::JSCRuntime::createObject(std::__1::shared_ptr<facebook::jsi::HostObject>)::HostObjectProxy::setProperty(OpaqueJSContext const*, OpaqueJSValue*, OpaqueJSString*, OpaqueJSValue const*, OpaqueJSValue const**) in libReact.a(JSCRuntime.o)
      facebook::jsc::JSCRuntime::createObject(std::__1::shared_ptr<facebook::jsi::HostObject>)::HostObjectProxy::getPropertyNames(OpaqueJSContext const*, OpaqueJSValue*, OpaqueJSPropertyNameAccumulator*) in libReact.a(JSCRuntime.o)
      facebook::jsc::JSCRuntime::createFunctionFromHostFunction(facebook::jsi::PropNameID const&, unsigned int, std::__1::function<facebook::jsi::Value (facebook::jsi::Runtime&, facebook::jsi::Value const&, facebook::jsi::Value const*, unsigned long)>)::HostFunctionMetadata::initialize(OpaqueJSContext const*, OpaqueJSValue*) in libReact.a(JSCRuntime.o)
      ...
  "_JSObjectGetProperty", referenced from:
      facebook::jsc::JSCRuntime::getProperty(facebook::jsi::Object const&, facebook::jsi::String const&) in libReact.a(JSCRuntime.o)
      facebook::jsc::JSCRuntime::getProperty(facebook::jsi::Object const&, facebook::jsi::PropNameID const&) in libReact.a(JSCRuntime.o)
      facebook::jsc::JSCRuntime::createFunctionFromHostFunction(facebook::jsi::PropNameID const&, unsigned int, std::__1::function<facebook::jsi::Value (facebook::jsi::Runtime&, facebook::jsi::Value const&, facebook::jsi::Value const*, unsigned long)>)::HostFunctionMetadata::initialize(OpaqueJSContext const*, OpaqueJSValue*) in libReact.a(JSCRuntime.o)
  "_JSObjectGetPropertyAtIndex", referenced from:
      facebook::jsc::JSCRuntime::getValueAtIndex(facebook::jsi::Array const&, unsigned long) in libReact.a(JSCRuntime.o)
  "_JSObjectGetPrototype", referenced from:
      facebook::jsc::JSCRuntime::createFunctionFromHostFunction(facebook::jsi::PropNameID const&, unsigned int, std::__1::function<facebook::jsi::Value (facebook::jsi::Runtime&, facebook::jsi::Value const&, facebook::jsi::Value const*, unsigned long)>)::HostFunctionMetadata::initialize(OpaqueJSContext const*, OpaqueJSValue*) in libReact.a(JSCRuntime.o)
  "_JSObjectHasProperty", referenced from:
      facebook::jsc::JSCRuntime::hasProperty(facebook::jsi::Object const&, facebook::jsi::String const&) in libReact.a(JSCRuntime.o)
      facebook::jsc::JSCRuntime::hasProperty(facebook::jsi::Object const&, facebook::jsi::PropNameID const&) in libReact.a(JSCRuntime.o)
  "_JSObjectIsFunction", referenced from:
      facebook::jsc::JSCRuntime::isFunction(facebook::jsi::Object const&) const in libReact.a(JSCRuntime.o)
  "_JSObjectMake", referenced from:
      facebook::jsc::JSCRuntime::makeObjectValue(OpaqueJSValue*) const in libReact.a(JSCRuntime.o)
      facebook::jsc::JSCRuntime::createObject(std::__1::shared_ptr<facebook::jsi::HostObject>) in libReact.a(JSCRuntime.o)
      facebook::jsc::JSCRuntime::createFunctionFromHostFunction(facebook::jsi::PropNameID const&, unsigned int, std::__1::function<facebook::jsi::Value (facebook::jsi::Runtime&, facebook::jsi::Value const&, facebook::jsi::Value const*, unsigned long)>) in libReact.a(JSCRuntime.o)
  "_JSObjectMakeArray", referenced from:
      facebook::jsc::JSCRuntime::createArray(unsigned long) in libReact.a(JSCRuntime.o)
  "_JSObjectSetPrivate", referenced from:
      facebook::jsc::JSCRuntime::createObject(std::__1::shared_ptr<facebook::jsi::HostObject>)::HostObjectProxy::finalize(OpaqueJSValue*) in libReact.a(JSCRuntime.o)
      facebook::jsc::JSCRuntime::createFunctionFromHostFunction(facebook::jsi::PropNameID const&, unsigned int, std::__1::function<facebook::jsi::Value (facebook::jsi::Runtime&, facebook::jsi::Value const&, facebook::jsi::Value const*, unsigned long)>)::HostFunctionMetadata::finalize(OpaqueJSValue*) in libReact.a(JSCRuntime.o)
  "_JSObjectSetProperty", referenced from:
      facebook::jsc::JSCRuntime::setPropertyValue(facebook::jsi::Object&, facebook::jsi::PropNameID const&, facebook::jsi::Value const&) in libReact.a(JSCRuntime.o)
      facebook::jsc::JSCRuntime::setPropertyValue(facebook::jsi::Object&, facebook::jsi::String const&, facebook::jsi::Value const&) in libReact.a(JSCRuntime.o)
      facebook::jsc::JSCRuntime::createArray(unsigned long) in libReact.a(JSCRuntime.o)
      facebook::jsc::JSCRuntime::createFunctionFromHostFunction(facebook::jsi::PropNameID const&, unsigned int, std::__1::function<facebook::jsi::Value (facebook::jsi::Runtime&, facebook::jsi::Value const&, facebook::jsi::Value const*, unsigned long)>)::HostFunctionMetadata::initialize(OpaqueJSContext const*, OpaqueJSValue*) in libReact.a(JSCRuntime.o)
  "_JSObjectSetPropertyAtIndex", referenced from:
      facebook::jsc::JSCRuntime::setValueAtIndexImpl(facebook::jsi::Array&, unsigned long, facebook::jsi::Value const&) in libReact.a(JSCRuntime.o)
  "_JSObjectSetPrototype", referenced from:
      facebook::jsc::JSCRuntime::createFunctionFromHostFunction(facebook::jsi::PropNameID const&, unsigned int, std::__1::function<facebook::jsi::Value (facebook::jsi::Runtime&, facebook::jsi::Value const&, facebook::jsi::Value const*, unsigned long)>)::HostFunctionMetadata::initialize(OpaqueJSContext const*, OpaqueJSValue*) in libReact.a(JSCRuntime.o)
  "_JSPropertyNameAccumulatorAddName", referenced from:
      facebook::jsc::JSCRuntime::createObject(std::__1::shared_ptr<facebook::jsi::HostObject>)::HostObjectProxy::getPropertyNames(OpaqueJSContext const*, OpaqueJSValue*, OpaqueJSPropertyNameAccumulator*) in libReact.a(JSCRuntime.o)
  "_JSPropertyNameArrayGetCount", referenced from:
      facebook::jsc::JSCRuntime::getPropertyNames(facebook::jsi::Object const&) in libReact.a(JSCRuntime.o)
  "_JSPropertyNameArrayGetNameAtIndex", referenced from:
      facebook::jsc::JSCRuntime::getPropertyNames(facebook::jsi::Object const&) in libReact.a(JSCRuntime.o)
  "_JSPropertyNameArrayRelease", referenced from:
      facebook::jsc::JSCRuntime::getPropertyNames(facebook::jsi::Object const&) in libReact.a(JSCRuntime.o)
  "_JSStringCreateWithUTF8CString", referenced from:
      facebook::jsc::JSCRuntime::evaluateJavaScript(std::__1::unique_ptr<facebook::jsi::Buffer const, std::__1::default_delete<facebook::jsi::Buffer const> >, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) in libReact.a(JSCRuntime.o)
      facebook::jsc::JSCRuntime::createPropNameIDFromAscii(char const*, unsigned long) in libReact.a(JSCRuntime.o)
      facebook::jsc::JSCRuntime::createPropNameIDFromUtf8(unsigned char const*, unsigned long) in libReact.a(JSCRuntime.o)
      facebook::jsc::JSCRuntime::createStringFromUtf8(unsigned char const*, unsigned long) in libReact.a(JSCRuntime.o)
      facebook::jsc::(anonymous namespace)::getLengthString() in libReact.a(JSCRuntime.o)
      facebook::jsc::(anonymous namespace)::getEmptyString() in libReact.a(JSCRuntime.o)
      facebook::jsc::(anonymous namespace)::getNameString() in libReact.a(JSCRuntime.o)
      ...
  "_JSStringGetMaximumUTF8CStringSize", referenced from:
      facebook::jsc::(anonymous namespace)::JSStringToSTLString(OpaqueJSString*) in libReact.a(JSCRuntime.o)
  "_JSStringGetUTF8CString", referenced from:
      facebook::jsc::(anonymous namespace)::JSStringToSTLString(OpaqueJSString*) in libReact.a(JSCRuntime.o)
  "_JSStringIsEqual", referenced from:
      facebook::jsc::JSCRuntime::compare(facebook::jsi::PropNameID const&, facebook::jsi::PropNameID const&) in libReact.a(JSCRuntime.o)
      facebook::jsc::JSCRuntime::strictEquals(facebook::jsi::String const&, facebook::jsi::String const&) const in libReact.a(JSCRuntime.o)
  "_JSStringRelease", referenced from:
      facebook::jsc::JSCRuntime::evaluateJavaScript(std::__1::unique_ptr<facebook::jsi::Buffer const, std::__1::default_delete<facebook::jsi::Buffer const> >, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) in libReact.a(JSCRuntime.o)
      facebook::jsc::JSCRuntime::JSCStringValue::invalidate() in libReact.a(JSCRuntime.o)
      facebook::jsc::JSCRuntime::createPropNameIDFromAscii(char const*, unsigned long) in libReact.a(JSCRuntime.o)
      facebook::jsc::JSCRuntime::createPropNameIDFromUtf8(unsigned char const*, unsigned long) in libReact.a(JSCRuntime.o)
      facebook::jsc::JSCRuntime::createValue(OpaqueJSValue const*) const in libReact.a(JSCRuntime.o)
      facebook::jsc::JSCRuntime::createFunctionFromHostFunction(facebook::jsi::PropNameID const&, unsigned int, std::__1::function<facebook::jsi::Value (facebook::jsi::Runtime&, facebook::jsi::Value const&, facebook::jsi::Value const*, unsigned long)>)::HostFunctionMetadata::initialize(OpaqueJSContext const*, OpaqueJSValue*) in libReact.a(JSCRuntime.o)
  "_JSStringRetain", referenced from:
      facebook::jsc::JSCRuntime::JSCStringValue::JSCStringValue(OpaqueJSString*, std::__1::atomic<long>&) in libReact.a(JSCRuntime.o)
      facebook::jsc::JSCRuntime::createFunctionFromHostFunction(facebook::jsi::PropNameID const&, unsigned int, std::__1::function<facebook::jsi::Value (facebook::jsi::Runtime&, facebook::jsi::Value const&, facebook::jsi::Value const*, unsigned long)>)::HostFunctionMetadata::HostFunctionMetadata(facebook::jsc::JSCRuntime*, std::__1::function<facebook::jsi::Value (facebook::jsi::Runtime&, facebook::jsi::Value const&, facebook::jsi::Value const*, unsigned long)>, unsigned int, OpaqueJSString*) in libReact.a(JSCRuntime.o)
  "_JSValueIsArray", referenced from:
      facebook::jsc::JSCRuntime::isArray(facebook::jsi::Object const&) const in libReact.a(JSCRuntime.o)
  "_JSValueIsBoolean", referenced from:
      facebook::jsc::JSCRuntime::createValue(OpaqueJSValue const*) const in libReact.a(JSCRuntime.o)
  "_JSValueIsInstanceOfConstructor", referenced from:
      facebook::jsc::JSCRuntime::instanceOf(facebook::jsi::Object const&, facebook::jsi::Function const&) in libReact.a(JSCRuntime.o)
  "_JSValueIsNull", referenced from:
      facebook::jsc::JSCRuntime::createValue(OpaqueJSValue const*) const in libReact.a(JSCRuntime.o)
  "_JSValueIsNumber", referenced from:
      facebook::jsc::JSCRuntime::createValue(OpaqueJSValue const*) const in libReact.a(JSCRuntime.o)
  "_JSValueIsObject", referenced from:
      facebook::jsc::JSCRuntime::createValue(OpaqueJSValue const*) const in libReact.a(JSCRuntime.o)
  "_JSValueIsObjectOfClass", referenced from:
      facebook::jsc::JSCRuntime::isHostObject(facebook::jsi::Object const&) const in libReact.a(JSCRuntime.o)
      facebook::jsc::JSCRuntime::isHostFunction(facebook::jsi::Function const&) const in libReact.a(JSCRuntime.o)
  "_JSValueIsString", referenced from:
      facebook::jsc::JSCRuntime::createValue(OpaqueJSValue const*) const in libReact.a(JSCRuntime.o)
  "_JSValueIsUndefined", referenced from:
      facebook::jsc::JSCRuntime::createValue(OpaqueJSValue const*) const in libReact.a(JSCRuntime.o)
  "_JSValueMakeBoolean", referenced from:
      facebook::jsc::JSCRuntime::valueRef(facebook::jsi::Value const&) in libReact.a(JSCRuntime.o)
  "_JSValueMakeNull", referenced from:
      facebook::jsc::JSCRuntime::valueRef(facebook::jsi::Value const&) in libReact.a(JSCRuntime.o)
  "_JSValueMakeNumber", referenced from:
      facebook::jsc::JSCRuntime::valueRef(facebook::jsi::Value const&) in libReact.a(JSCRuntime.o)
      facebook::jsc::JSCRuntime::createArray(unsigned long) in libReact.a(JSCRuntime.o)
      facebook::jsc::JSCRuntime::createFunctionFromHostFunction(facebook::jsi::PropNameID const&, unsigned int, std::__1::function<facebook::jsi::Value (facebook::jsi::Runtime&, facebook::jsi::Value const&, facebook::jsi::Value const*, unsigned long)>)::HostFunctionMetadata::initialize(OpaqueJSContext const*, OpaqueJSValue*) in libReact.a(JSCRuntime.o)
  "_JSValueMakeString", referenced from:
      facebook::jsc::JSCRuntime::valueRef(facebook::jsi::Value const&) in libReact.a(JSCRuntime.o)
      facebook::jsc::JSCRuntime::createFunctionFromHostFunction(facebook::jsi::PropNameID const&, unsigned int, std::__1::function<facebook::jsi::Value (facebook::jsi::Runtime&, facebook::jsi::Value const&, facebook::jsi::Value const*, unsigned long)>)::HostFunctionMetadata::initialize(OpaqueJSContext const*, OpaqueJSValue*) in libReact.a(JSCRuntime.o)
  "_JSValueMakeUndefined", referenced from:
      facebook::jsc::JSCRuntime::valueRef(facebook::jsi::Value const&) in libReact.a(JSCRuntime.o)
      facebook::jsc::JSCRuntime::createObject(std::__1::shared_ptr<facebook::jsi::HostObject>)::HostObjectProxy::getProperty(OpaqueJSContext const*, OpaqueJSValue*, OpaqueJSString*, OpaqueJSValue const**) in libReact.a(JSCRuntime.o)
      facebook::jsc::JSCRuntime::createFunctionFromHostFunction(facebook::jsi::PropNameID const&, unsigned int, std::__1::function<facebook::jsi::Value (facebook::jsi::Runtime&, facebook::jsi::Value const&, facebook::jsi::Value const*, unsigned long)>)::HostFunctionMetadata::call(OpaqueJSContext const*, OpaqueJSValue*, OpaqueJSValue*, unsigned long, OpaqueJSValue const* const*, OpaqueJSValue const**) in libReact.a(JSCRuntime.o)
  "_JSValueProtect", referenced from:
      facebook::jsc::JSCRuntime::JSCObjectValue::JSCObjectValue(OpaqueJSContext*, std::__1::atomic<bool> const&, OpaqueJSValue*, std::__1::atomic<long>&) in libReact.a(JSCRuntime.o)
  "_JSValueToBoolean", referenced from:
      facebook::jsc::JSCRuntime::createValue(OpaqueJSValue const*) const in libReact.a(JSCRuntime.o)
  "_JSValueToNumber", referenced from:
      facebook::jsc::JSCRuntime::createValue(OpaqueJSValue const*) const in libReact.a(JSCRuntime.o)
  "_JSValueToObject", referenced from:
      facebook::jsc::JSCRuntime::createValue(OpaqueJSValue const*) const in libReact.a(JSCRuntime.o)
      facebook::jsc::JSCRuntime::createFunctionFromHostFunction(facebook::jsi::PropNameID const&, unsigned int, std::__1::function<facebook::jsi::Value (facebook::jsi::Runtime&, facebook::jsi::Value const&, facebook::jsi::Value const*, unsigned long)>)::HostFunctionMetadata::initialize(OpaqueJSContext const*, OpaqueJSValue*) in libReact.a(JSCRuntime.o)
  "_JSValueToStringCopy", referenced from:
      facebook::jsc::JSCRuntime::createValue(OpaqueJSValue const*) const in libReact.a(JSCRuntime.o)
  "_JSValueUnprotect", referenced from:
      facebook::jsc::JSCRuntime::JSCObjectValue::invalidate() in libReact.a(JSCRuntime.o)
  "_kJSClassDefinitionEmpty", referenced from:
      facebook::jsc::JSCRuntime::createObject(std::__1::shared_ptr<facebook::jsi::HostObject>)::$_0::operator()() const in libReact.a(JSCRuntime.o)
      facebook::jsc::JSCRuntime::createFunctionFromHostFunction(facebook::jsi::PropNameID const&, unsigned int, std::__1::function<facebook::jsi::Value (facebook::jsi::Runtime&, facebook::jsi::Value const&, facebook::jsi::Value const*, unsigned long)>)::$_1::operator()() const in libReact.a(JSCRuntime.o)
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

**Comment by radex** [comment]
@blainekasten I think you didn't link your project with JavaScriptCore.framework

**Comment by grabbou** [comment]
We have just released a patch to CLI to fix @Salakar issues. No changes required, just reinstall your Node modules.

**Comment by grabbou** [comment]
@Kennytian make your `start` script: `react-native start` instead of `node ./node_modules/react-native/local-cli/cli.js`.

This was an unintended breaking change and we are shipping a 0.59.0-rc.1 right now with a fix.

**Comment by tamirrab** [comment]
Duplicate of #23453

**Comment by tamirrab** [comment]
#23453

**Comment by IljaDaderko** [comment]
Hey guys,

Upgraded my project from `0.58.4` to `0.59.0-rc.0`. One issue pops up when building iOS
> project/Pods/Folly/folly/portability/Config.h:20:10: 'folly/folly-config.h' file not found

This is after cleaning build folder, reseting packager cache etc... also, project includes react native through Cocoapods (not sure if relevant) i.e.
```
  pod 'yoga', :path => '../../node_modules/react-native/ReactCommon/yoga'
  pod 'DoubleConversion', :podspec => '../../node_modules/react-native/third-party-podspecs/DoubleConversion.podspec'
  pod 'glog', :podspec => '../../node_modules/react-native/third-party-podspecs/glog.podspec'
  pod 'Folly', :podspec => '../../node_modules/react-native/third-party-podspecs/Folly.podspec'
  pod 'React', :path => '../../node_modules/react-native', :subspecs => [
    'Core',
    'CxxBridge',
    'DevSupport',
    'RCTAnimation',
    'RCTImage',
    'RCTLinkingIOS',
    'RCTNetwork',
    'RCTText',
    'RCTWebSocket'
  ]
```

**Comment by Salakar** [comment]
@IljaDaderko same here;

![image](https://user-images.githubusercontent.com/5347038/52798058-1991cf00-306f-11e9-9acb-8b30f33154a9.png)

**Comment by danilobuerger** [comment]
When will rc1 be available via npm?

**Comment by grabbou** [comment]
We are having some issues with the CI so can’t guarantee any dates. We are
working on it.

**Comment by hramos** [comment]
Heads up, there's [a bug in Hooks](https://github.com/facebook/react/issues/14849) that would be nice to get fixed before 0.59 stable is cut. It looks like a fix for this might land next week if all goes well, so we should be able to get it into the 0.59-RC before it's too late.

**Comment by yurykorzun** [comment]
It looks like React.podspec is broken. Core subspec is missing Folly headers in search paths.

**Comment by ltcaosj** [comment]
**react-native link** command does not work.
<img width="1642" alt="rn059_link1" src="https://user-images.githubusercontent.com/5376380/52833602-1ac70880-3110-11e9-8d9e-9c405f98bf71.png">
<img width="979" alt="rn059_link2" src="https://user-images.githubusercontent.com/5376380/52833607-20bce980-3110-11e9-88a7-6137f710fdca.png">

**Comment by dulmandakh** [comment]
created PR to fix link command https://github.com/react-native-community/react-native-cli/pull/170

**Comment by ltcaosj** [comment]
unlink command is also not working. 
<img width="743" alt="rn59_link3" src="https://user-images.githubusercontent.com/5376380/52840468-f711bb80-312b-11e9-9bb3-65b9676e3132.png">

The assets link also does not work.
"rnpm": {
    "assets": [
      "./src/assets/fonts/"
    ]
  }

**Comment by grabbou** [comment]
CLI issues fixed in `1.2.3` release. Please reinstall your `node_modules` to get a fix automatically.

**Comment by dulmandakh** [comment]
there is a PR to fix Android crash issue https://github.com/facebook/react-native/pull/23470. please include this in the next RC

**Comment by swensonmattias** [comment]
> CLI issues fixed in `1.2.3` release. Please reinstall your `node_modules` to get a fix automatically.

Even after `yarn cache clean` I still get v1.2.2 when reinstalling. Do I need to specify the version somewhere?
I don't see v1.2.3 on NPM.

**Comment by dulmandakh** [comment]
cherry pick:
- bump Android Plugin to 3.3.1, to silence warning in Android Studio - https://github.com/facebook/react-native/commit/da5b5d2fa134aa09dda4a620be9fa4d3d419201f

**Comment by guhungry** [comment]
> > CLI issues fixed in `1.2.3` release. Please reinstall your `node_modules` to get a fix automatically.
> 
> Even after `yarn cache clean` I still get v1.2.2 when reinstalling. Do I need to specify the version somewhere?
> I don't see v1.2.3 on NPM.

delete `yarn.lock` then `yarn install`

**Comment by ChrisEdson** [comment]
1.2.3 is not yet on NPM

**Comment by grabbou** [comment]
@swensonmattias my bad, try it out now. Should be good.

**Comment by ltcaosj** [comment]
With the latest master, the **react-native link** command is working fine now but the **react-native unlink command** still does not work.
<img width="633" alt="rn59_link4" src="https://user-images.githubusercontent.com/5376380/52864964-0154a980-316e-11e9-9319-0973952667e9.png">

**Comment by yilmarnoriega93** [comment]
problem to react-native start 
![whatsapp image 2019-02-15 at 10 35 57 am](https://user-images.githubusercontent.com/47668219/52866892-a8edbf80-310d-11e9-94d9-d3193a3a00c9.jpeg)

**Comment by grabbou** [comment]
React Native 0.59.0-rc.1 is out, contains few Android fixes (as requested by @dulmandakh earlier) and CocoaPods fix.

**Comment by yilmarnoriega93** [comment]
> React Native 0.59.0-rc.1 is out, contains few Android fixes (as requested by @dulmandakh earlier) and CocoaPods fix.

@grabbou update react-native-cli? I still have the same error. (Unknown)

**Comment by IljaDaderko** [comment]
I can still see same error as in https://github.com/react-native-community/react-native-releases/issues/79#issuecomment-463628317 within rc1. Was cocoapods fix for that or something else? Perhaps @Salakar @yurykorzun or @Bardiamist can confirm?

**Comment by nar213** [comment]
Please can this be considered for cherry-picking https://github.com/facebook/react-native/commit/dd209bb7891ed5f05b96a9922c7b0e39bf3ac9e9 there have been iOS web socket issues in all versions since 0.57

**Comment by yurykorzun** [comment]
> I can still see same error as in [#79 (comment)](https://github.com/react-native-community/react-native-releases/issues/79#issuecomment-463628317) within rc1. Was cocoapods fix for that or something else? Perhaps @Salakar @yurykorzun or @Bardiamist can confirm?

I cannot build 0.59 rc1 using a podspec, the same Folly compiler issue remains. There were no changes committed to React.podspec.

**Comment by hramos** [comment]
There's an issue in 0.59 we should fix before hitting stable: https://github.com/facebook/react-native/issues/23235

**Comment by fkgozali** [comment]
> There's an issue in 0.59 we should fix before hitting stable: [facebook/react-native#23235](https://github.com/facebook/react-native/issues/23235)

Let's pick this commit: https://github.com/facebook/react-native/commit/a9dd828c68338dbf0e55ffa1838bf8ff574f317d

**Comment by gaodeng** [comment]
Would be nice to include https://github.com/facebook/react-native/commit/f15145639dab1e8d7a1c79a127b7d45c91d025a8

**Comment by spyr0s** [comment]
I have this error using 0.59.0-rc.1
```
FAILURE: Build failed with an exception.

* Where:
Script '.../node_modules\react-native\react.gradle' line: 95

* What went wrong:
A problem occurred configuring project ':app'.
> Could not get unknown property 'mergeResourcesProvider' for object of type com.android.build.gradle.internal.api.ApplicationVariantImpl.
```

**Comment by Federkun** [comment]
@spyr0s , refer to https://github.com/react-native-community/react-native-releases/issues/79#issuecomment-463448276

**Comment by Salakar** [comment]
> > I can still see same error as in [#79 (comment)](https://github.com/react-native-community/react-native-releases/issues/79#issuecomment-463628317) within rc1. Was cocoapods fix for that or something else? Perhaps @Salakar @yurykorzun or @Bardiamist can confirm?
> 
> I cannot build 0.59 rc1 using a podspec, the same Folly compiler issue remains. There were no changes committed to React.podspec. I added "$(PODS_ROOT)/Folly/folly"- recursive to header search paths and then it produced another compiler error in fishook. It says cstdlib is missing.
> 
> I am not sure what needs to be changed in podspec to match settings in React.xcodeproj.

Confirmed still the same issue here on rc1

**Comment by Salakar** [comment]
@yurykorzun @IljaDaderko I have a fix for the folly issue. I will send up a PR shortly if you could apply it locally let me know

**Comment by IljaDaderko** [comment]
Applied @Salakar 's PR and folly issue is resolved 👍

**Comment by grabbou** [comment]
Going to wait for @Salakar PR to land and will release 0.59.0-rc.2 with fixes mentioned in this thread since the last release.

**Comment by Sharlaan** [comment]
Not sure where to report htis issue, but trying RN 0.59-rc.1 (and React 16.8.2) on Win10 led me to a path resolution problem :

![image](https://user-images.githubusercontent.com/2692027/52916136-38e16400-32dc-11e9-8679-ce36b5196217.png)

I never got this issue with previous versions.

**Comment by Salakar** [comment]
@grabbou PR has landed 🎉

**Comment by grabbou** [comment]
Cherry-picked requested commits and pushed the release for tests. Will try to publish it soon.

Note: I am marking comments that talk about resolved issues as "resolved" to keep visibility on the issues that we haven't fixed yet or known problems (e.g. upgrading Android SDK).

**Comment by grabbou** [comment]
0.59.0-rc.1 is on CircleCI, should be live soon. 

Cherry-picked 3 commits, including CocoaPods fix. See them [here](https://github.com/facebook/react-native/commits/0.59-stable) 

CLI work is happening in a different repository, reinstalling `node_modules` should do the trick.

**Comment by thymikee** [comment]
@Sharlaan can you translate this error to English? I'm not sure what's wrong there.

**Comment by Sharlaan** [comment]
Sorry i thought it would be self-explanatory.

Basically it says that it fails to execute what it think to be a cli command, because of the space in the path `D://My          projects/..`

**Comment by thymikee** [comment]
@Sharlaan Thanks! Which version of Node are you running?

**Comment by Sharlaan** [comment]
last, v11.10

**Comment by thymikee** [comment]
Did it work for you earlier? This code hasn't really changed since extraction

**Comment by Sharlaan** [comment]
after i search or the previous version of RN i were using i found out it were 0.58.4 but most importantly, the repo were manually settled...
... unlike here i tested using directly `expo init`

**Comment by thymikee** [comment]
I don't have a Windows machine, and paths with spaces work on UNIX. I'd appreciate your help on that. The relevant code is here: https://github.com/react-native-community/react-native-cli/blob/760d384ae199ccd6552805459c195ed5aa3969fc/packages/cli/src/cliEntry.js#L142-L146

It may be an Expo issue as you say, but it's possible that we can make it easier for them on our side.

**Comment by Sharlaan** [comment]
ok i made another fresh test which worked flawlessly :
![image](https://user-images.githubusercontent.com/2692027/52970962-fa1fdc80-33b5-11e9-9a61-9e505f449be6.png)

... but after upgrading, red screen (LAN mode) :
![image](https://user-images.githubusercontent.com/2692027/52971175-cbeecc80-33b6-11e9-8f45-e8ba349985c8.png)

Honestly i'm clueless there :s

**Comment by Sharlaan** [comment]
I made a few research about `childProcess.execFile` and found this in Node documentation :
https://nodejs.org/api/child_process.html#child_process_spawning_bat_and_cmd_files_on_windows
(especially last example)

Any space in the argument passed to `childProcess.execFileSync` tells it there is a command executable.
So maybe the solution would be to enclose the whole path string in quotes ?

**Comment by thymikee** [comment]
@Sharlaan feel free to try it! :)

**Comment by grabbou** [comment]
0.59.0-rc.2 is out on npm now.

**Comment by kelset** [comment]
@thymikee @Sharlaan can you please move your conversation to an issue in the main repo? So that we can avoid polluting this issue

**Comment by ltcaosj** [comment]
**- Cannot run DEBUG on a real iOS device**
**- Cannot even run RN app on a real Android device**
**Reproduce:** Just init a new RN app ---> got those issues
<img width="650" alt="device1" src="https://user-images.githubusercontent.com/5376380/53018270-8567a380-3484-11e9-95da-453af3dc8e4f.png">
<img width="1018" alt="device2" src="https://user-images.githubusercontent.com/5376380/53018283-8ac4ee00-3484-11e9-8243-320afd285251.png">

**Comment by kelset** [comment]
@ltcaosj please open a dedicated issue on the main repo following the template.

**Comment by ltcaosj** [comment]
@kelset , I thought the issue is related to 0.59.x release, so I put it here. Let me put it in the main repo then.

**Comment by akshetpandey** [comment]
@grabbou : https://github.com/react-native-community/react-native-releases/issues/79#issuecomment-460410565 isn't resolved and I feel like its still a blocker for 0.59 release. Its being discussed here: https://github.com/react-native-community/jsc-android-buildscripts/issues/82

**Comment by danilobuerger** [comment]
Please cherry pick:
Map TextInput textContentType strings to Objective-C constants https://github.com/facebook/react-native/commit/a89fe4165c2a331a9d88636d89a5a48151ab8660

**Comment by dulmandakh** [comment]
Cherry picks:
- SYSTEM_ALERT_WINDOW permission only in debug builds - https://github.com/facebook/react-native/commit/84a2fb0a4a67cd9dc37cf4e5bab814f25181cfb7
- ReactTextView extends AppCompatTextView - https://github.com/facebook/react-native/commit/ba0c3ffd5b46963a8bb27b40eb396965535cd927

**Comment by ziqiDev** [comment]
when will update the version of gradle  to 5.0?

**Comment by dulmandakh** [comment]
@ziqiDev it's landed in master, and will be a part of 0.60 release
https://github.com/facebook/react-native/commit/b4017a9923b09fed4b693a8e4cfadd30ce34c88d

**Comment by ltcaosj** [comment]
I just installed the latest release of 0.59.x : The **Live Reload does not work** - I need to reinstall the app every time making a code change.

**Comment by t3chnoboy** [comment]
@ltcaosj I have the same issue. The development server is not exposed to the local network.
As a workaround you can start the dev server like this: `react-native start --host 0.0.0.0`

**Comment by ltcaosj** [comment]
thanks @t3chnoboy , your workaround solution works ;)

**Comment by kelset** [comment]
@ltcaosj @t3chnoboy it looks like it's a CLI issue, can you please open one in the [dedicated repo](https://github.com/react-native-community/react-native-cli/issues) so that it can be investigated?

**Comment by ltcaosj** [comment]
> @ltcaosj @t3chnoboy it looks like it's a CLI issue, can you please open one in the [dedicated repo](https://github.com/react-native-community/react-native-cli/issues) so that it can be investigated?

https://github.com/facebook/react-native/issues/23589

**Comment by sunnylqm** [comment]
Please cherry pick:
https://github.com/facebook/react-native/commit/d9489c4e9c646b79025f07635b840e9974be8cd5

**Comment by dulmandakh** [comment]
Cherry picks:
- Add Nonnull annotations to ViewManager - https://github.com/facebook/react-native/commit/0b4f74176c9a2d21d0bcf9c497b7ce7440f6a920
- Use existing character set in POST body when possible - https://github.com/facebook/react-native/commit/0d5aebbd9ac92a90ec7ab1426ed92dd22ae8c736
- Fix IllegalArgumentException when creating CookieManager - https://github.com/facebook/react-native/commit/cda8171af30815edfa331e07d1bbf605f0926303
- Prevent okhttp from adding ;charset=utf8 to ContentType Header - https://github.com/facebook/react-native/commit/4a807761a4aca9e551ff2cee8ca18a2450fb11ca
- Fix two bugs with Location when not using ACCESS_FINE_LOCATION -https://github.com/facebook/react-native/commit/f32dc635467a2e93371f0cf2e40b07a712349288

**Comment by tomduncalf** [comment]
Hey, wanted to report an issue I've found with 0.59rc2.

Our project can't use the `-ObjC` linker flag to discover/load all symbols as it causes "duplicate symbol" errors for some of the static libraries we use (`ObjC` is all-or-nothing flag, so we can't opt these out of it). As RN depends on this flag, the alternative is to use `-force_load` for each library that needs to be loaded in this way and specify the path to the compiled (`.a`) library.

With 0.58 this compiles fine, but with 0.59rc2 it results in "duplicate symbol" errors for some Yoga references in `libReact.a`:

![screenshot 2019-02-25 at 18 10 14](https://user-images.githubusercontent.com/5458070/53359194-1ed20200-392a-11e9-9484-a5c4f5fee8c4.png)

I've created a repro case: https://github.com/tomduncalf/rn0.59-force-load-bug - in the interest of keeping the project simple, the path to `libReact.a` needs to be manually changed in "Linker Flags", you can discover it by going to `~/Library/Developer/Xcode/DerivedData` and looking for the directory starting `forceLoad59rc2-....` then replacing the existing one with this path (feels like there should be a `$(..)` variable pointing to this location but I've never found it!).

Please let me know if you'd like more help looking into this issue or reproducing it - this would be a blocking issue for our project as we can't compile with `-ObjC`.

**Comment by FLGMwt** [comment]
Can we please cherry-pick: https://github.com/facebook/react-native/commit/b9d3743cda95d1f475dbec8f6d72935941519deb ?

Not a regression or security fix, but it addresses a pretty big long-standing gap w/ a11y on Android.

**Comment by hramos** [comment]
@tomduncalf can you open a new issue in https://github.com/facebook/react-native and try and make it clear it's present in the 0.59 RC? It will be easier to track that way.

**Comment by tomduncalf** [comment]
@hramos sure, https://github.com/facebook/react-native/issues/23645 raised. Thanks!

**Comment by ericlewis** [comment]
Can we cherry pick: https://github.com/facebook/react-native/commit/8ce3c1b43edd47191c8e5ee8432c58f6e93dfca7

Solves an issue we have had since 2016.

**Comment by hramos** [comment]
Please cherry pick https://github.com/facebook/react-native/commit/d4ce8463a853a29cc977fcbf390b43940f521bd2. This commit bumps React to v16.8.3 and updates the React Native renderer accordingly.

**Comment by akshetpandey** [comment]
https://github.com/react-native-community/react-native-releases/issues/79#issuecomment-465368826 is resolved.

**Comment by grabbou** [comment]
@akshetpandey - do you mean we are good to go? My understanding from https://github.com/react-native-community/jsc-android-buildscripts/issues/82 is that it's not a blocker for 0.59 anymore.

**Comment by akshetpandey** [comment]
Yes, its not a blocker for 0.59. Looks like 0.59 improves on 0.58 behavior.

**Comment by grabbou** [comment]
Awesome! Going to "resolve" all the comments then to clean it up! Thank you so much for being active in that topic here and in that issue!

**Comment by grabbou** [comment]
0.59.0-rc.3 is out

**Comment by zenz** [comment]
should upgrade the .flowconfig to version 0.93.0
it seems this version(flow-bin@0.93.0) solved the 'flow check dosen't work under windows' issue. 
https://github.com/facebook/react-native/issues/21311

**Comment by StasDoskalenko** [comment]
Hey guys! Please add this to 0.59 
https://github.com/facebook/react-native/commit/f3e5cce4745c0ad9a5c697be772757a03e15edc5

**Comment by vdlindenmark** [comment]
I'm getting this error when trying to run my app in release mode (iOS, Xcode, Cocoapods)

```
Undefined symbols for architecture arm64:
  "_RCTSetLogFunction", referenced from:
      -[AppName testRendersWelcomeScreen] in AppName
ld: symbol(s) not found for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

**Comment by hramos** [comment]
@vdlindenmark can you open a new issue in the [React Native repository](https://github.com/facebook/react-native/issues) instead? Just make sure to make it clear it affects 0.59.0-rc.3 and to include a link back to this issue so we can keep track of it. I'll hide your comment to make sure we keep this thread to cherry-pick requests.

**Comment by hramos** [comment]
@StasDoskalenko that is already part of 0.59, and it's one of the tentpole features for this release.

**Comment by hramos** [comment]
@zenz please stick to cherry-pick requests for commits that are already in React Native master. Please do open a new issue in the [React Native repository](https://github.com/facebook/react-native/issues) if you have not yet, and link back to this issue. That way we can keep track of work that might benefit this release. Thanks!

**Comment by shawnthye** [comment]
Hi, will this https://github.com/facebook/react-native/pull/23430 come into the next patch?

**Comment by vdlindenmark** [comment]
> @vdlindenmark can you open a new issue in the [React Native repository](https://github.com/facebook/react-native/issues) instead? Just make sure to make it clear it affects 0.59.0-rc.3 and to include a link back to this issue so we can keep track of it. I'll hide your comment to make sure we keep this thread to cherry-pick requests.

I'm sorry. You could ignore my message. I've didn't configure the cocoapods right, discovered that this morning

**Comment by grabbou** [comment]
Cherry pick
https://github.com/facebook/react-native/commit/e10d984c437187b2cfdb364248c93a261d3f5e46
to
upgrade Jest from an alpha release.

**Comment by satyen-progton** [comment]
Please use android:usesCleartextTraffic="true" in AndroidManifest to allow development server run on android 9.0 devices

**Comment by hramos** [comment]
@satyen-progton please send a pull request to the core RN repository. We are only handling cherry pick requests in this thread.

**Comment by kelset** [comment]
@satyen-progton @hramos - actually @Salakar already added this to the docs -> https://facebook.github.io/react-native/docs/integration-with-existing-apps#network-security-config-api-level-28 (select the Android tab and mid way you'll see a dedicated section).

May be worth highlighting in the changelog for 0.59 cc @turnrye

**Comment by furkancelik** [comment]
ios or android file cannot be recreated. (dont work react-native eject)

I get the following error when I run the react-native eject command when I want to delete and re-create the iOS or Android file.

<img width="568" alt="ekran resmi 2019-03-01 23 18 34" src="https://user-images.githubusercontent.com/6446228/53687164-4f5ad700-3d42-11e9-9ac5-9d85136be4c5.png">

**Comment by kelset** [comment]
@furkancelik please open a dedicated issue in the correct repo (probably the RNCli one).

**Comment by OlivierFreyssinet** [comment]
Hi! Is this https://github.com/facebook/react-native/commit/58c3a4c078c066038baa9a961afae498557a1e18 going to be in 0.59 ?

**Comment by brunolemos** [comment]
https://github.com/facebook/react-native/commit/22b3528e7b5679ca34436bc032e6f1f1630d3c07
So I can drop my fork/patches.

**Comment by sunnylqm** [comment]
Hi can you cherry pick this one https://github.com/facebook/react-native/commit/d2153fc58d825006076a3fce12e0f7eb84479132

**Comment by tijp** [comment]
Hello! Can https://github.com/facebook/react-native/pull/23446 be cherry picked for 0.59?

**Comment by ChrisEdson** [comment]
Out of interest, how close are we getting to a stable release of 0.59? It would be good purely because we're sort of in a holding pattern for developing things with hooks.

**Comment by grabbou** [comment]
We are waiting for the changelog and blog post announcing 0.59 and we will
be making it stable. If it's not ready today, then, it's going to be Monday
to avoid releasing on Friday.

**Comment by matt-block** [comment]
So [facebook/react-native#23645](https://github.com/facebook/react-native/issues/23645) is not a release blocker anymore? Has it been solved in RC 3?

**Comment by grabbou** [comment]
Didn't know about that one, thanks for cross-posting.

**Comment by hramos** [comment]
The release blocker label tracks issues introduced during the RC period, and is meant to help us avoid shipping a regression in a stable release. It is not intended for these issues to prevent new RC patches from being created.

**Comment by matt-block** [comment]
@hramos Then I simply misinterpreted its meaning, thanks for clarifying!

**Comment by grabbou** [comment]
We decided to wait with 0.59 stable release till next week (Monday/Tuesday) till there's more context on the aforementioned regression. We want to be sure whether it's a major that should be fixed before a stable release happen or a minor that we can disclose in the changelog as a "known issue".

**Comment by cjroth** [comment]
Ahoy!

1. Excited about 0.59 stable.
2. Given that it's going to be released Mon/Tue, looks like this memory leak will have to wait until 0.60? :( https://github.com/facebook/react-native/issues/23801

Seems like it should be fairly high priority.

**Comment by kelset** [comment]
@cjroth given that, as you posted in the issue, this seems to be a long lasting issue with RN (going as far as 0.57 at least) it won’t be considered a blocker for the release on 0.59.0. OFC any help you can provide over there in terms of debugging it and PRs will be super appreciated, and we’ll see how we can prioritize it against the current roadmap & pipeline.

Also, the PR with the fix of the blocker mentioned above is being imported, so hopefully we can still release 0.59.0 next week.

**Comment by cjroth** [comment]
@kelset sweet, excited about 0.59.0, thanks for the reply. I wish I had more time and was better at the native side of things. If I can find some time I might poke around but I already spent a few days tracking it down so not sure if I'll be able to find the time. Just wanted to make sure some folks in here were aware of it!

**Comment by grabbou** [comment]
Turned out the issue was not really major. Anyway, since the PR is already being imported (https://github.com/facebook/react-native/pull/23823), we're going to wait for it to land and then, cut the RC.

I am hoping that means "today"

**Comment by krunalsshah** [comment]
Looks `OkHttpClientProvider.replaceOkHttpClient(myHttpClient);` is a deprecated change [Android]

What is the suggestion if we want to replace the default OkHttpClient used by react native?

**Comment by grabbou** [comment]
Closing this one as 0.59.0 is out. Let's continue discussions in a separate issue.
